src/Controller/SearchAction.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Controller;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Admin\BreadcrumbsBuilderInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Search\SearchHandler;
+use Sonata\AdminBundle\Templating\TemplateRegistryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+final class SearchAction extends Controller
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var SearchHandler
+     */
+    private $searchHandler;
+
+    /**
+     * @var TemplateRegistryInterface
+     */
+    private $templateRegistry;
+
+    /**
+     * @var BreadcrumbsBuilderInterface
+     */
+    private $breadcrumbsBuilder;
+
+    public function __construct(
+        Pool $pool,
+        SearchHandler $searchHandler,
+        TemplateRegistryInterface $templateRegistry,
+        BreadcrumbsBuilderInterface $breadcrumbsBuilder
+    ) {
+        $this->pool = $pool;
+        $this->searchHandler = $searchHandler;
+        $this->templateRegistry = $templateRegistry;
+        $this->breadcrumbsBuilder = $breadcrumbsBuilder;
+    }
+
+    /**
+     * The search action first render an empty page, if the query is set, then the template generates
+     * some ajax request to retrieve results for each admin. The Ajax query returns a JSON response.
+     *
+     * @throws \RuntimeException
+     *
+     * @return JsonResponse|Response
+     */
+    public function __invoke(Request $request)
+    {
+        if ($request->get('admin') && $request->isXmlHttpRequest()) {
```

Comment:
What about inversing if and putting the default return instead of this.

Reply:
I just did this, good call :)